src/features/packages/packageRepository.ts

```diff
@@ -0,0 +1,77 @@
+import { Package } from "./package";
+import { promises as fs } from "fs";
+import path from "path";
+import FormData from "form-data";
+import { OverwriteMode } from "../overwriteMode";
+import { ListArgs } from "../basicRepository";
+import { Client } from "../../client";
+import { spaceScopedRoutePrefix } from "../../spaceScopedRoutePrefix";
+import { ResourceCollection } from "../../resourceCollection";
+import { resolveSpaceId } from "../../spaceResolver";
+
+type PackagesListArgs = {
+    nuGetPackageId?: string;
+    filter?: string;
+    latest?: boolean;
+    includeNotes?: boolean;
+} & ListArgs;
+
+export class PackageRepository {
+    private client: Client;
+    private spaceName: string;
+
+    constructor(client: Client, spaceName: string) {
+        this.client = client;
+        this.spaceName = spaceName;
+    }
+
+    async get(packageId: string): Promise<Package> {
+        if (!packageId) {
+            throw new Error("Package Id was not provided");
+        }
+        const response = await this.client.request<Package>(`${spaceScopedRoutePrefix}/packages{/packageId}`, {
+            spaceName: this.spaceName,
+            packageId,
+        });
+        return response;
+    }
+
+    async list(args?: PackagesListArgs): Promise<ResourceCollection<Package>> {
+        const response = await this.client.request<ResourceCollection<Package>>(
+            `${spaceScopedRoutePrefix}/packages{/id}{?nuGetPackageId,filter,latest,skip,take,includeNotes}`,
+            {
+                spaceName: this.spaceName,
+                ...args,
+            }
+        );
+        return response;
+    }
+
+    async push(packages: string[], overwriteMode: OverwriteMode = OverwriteMode.FailIfExists): Promise<void> {
+        const spaceId = await resolveSpaceId(this.client, this.spaceName);
+
+        const tasks: Promise<void>[] = [];
+
+        for (const packagePath of packages) {
+            tasks.push(this.packageUpload(spaceId, packagePath, overwriteMode));
+        }
+
+        await Promise.all(tasks);

```

Comment:
Same as before, should we be using allSettled instead of all?

Reply:
Fixed